src/langcheck/metrics/de/_translation.py

```diff
@@ -0,0 +1,63 @@
+from math import floor
+
+from nltk.tokenize import sent_tokenize
+from transformers.pipelines import pipeline
+
+
+class Translate:
+    '''Translation class based on HuggingFace's translation pipeline.'''
+
+    def __init__(self, model_name: str) -> None:
+        '''
+        Initialize the Translation class with given parameters.
+
+        Args:
+            model_name: The name of the model to use for translation
+        '''
+        self._translation_pipeline = pipeline("translation",
+                                              model=model_name,
+                                              tokenizer=model_name)
+        self._max_length = self._translation_pipeline.model.config.max_length
+
+    def _translate(self, texts: str) -> str:
+        '''Translate the texts using the translation pipeline.
+        It splits the texts into blocks and translates each block separately,
+        avoiding problems with long texts.
+        Args:
+            texts: The texts to translate
+        Returns:
+            The translated texts
+        '''
+        tokenization = self._translation_pipeline.tokenizer(
+            texts, return_tensors="pt")  # type: ignore
+        if tokenization.input_ids.shape[1] > (self._max_length / 2):
```

Comment:
At max length it fails badly, and if running on CUDA, very badly. Since tokenization was not a perfect measure (don't know why), I went for 1/2 length instead of full length. It will still hit only if we have texts of more than ~300 tokens, so longish...

Reply:
Got it!